docs/gettingstarted/structuring.md

```diff
@@ -1,124 +1,81 @@
 
-A common subpackage strategy has packages for `common` and `client` code, which is code that can be run on both server/client and only client, respectively. Inside the `common` package would go things like Items, Blocks, and Block Entities (which can each, in turn, be another subpackage). Things like Screens and Renderers would go inside the `client` package.
+#### Client, Server, and Data Packages
 
-!!! note
-    This package style is only a suggestion, though it is a commonly used style. Feel free to use your own packaging system.
+In general, code only for a given side or runtime should be isolated from the other classes in a separate subpackage. For example, code related to [data generation][datagen] should go in a `data` package while code only on the dedicated server should go in a `server` package.
 
-By keeping your code in clean subpackages, you can grow your mod much more organically.
+However, it is highly recommended that [client-only code][sides] should be isolated in a `client` subpackage. This is because dedicated servers have no access to any of the client-only packages in Minecraft. As such, having a dedicated package would provide a decent sanity check to verify you are not reaching across sides within your mod.
 
 Class Naming Schemes
 --------------------
 
-A common class naming scheme allows easier deciphering of what a class is, and it also makes it easier for someone developing with your mod to find things.
+A common class naming scheme makes it easier to decipher the purpose of the class or to easily locate specific classes.
+
+Classes are commonly suffixed with its type, for example:
 
-For Example:
+* An `Item` called `PowerRing` -> `PowerRingItem`.
+* A `Block` called `NotDirt` -> `NotDirtBlock`.
+* A menu for an `Oven` -> `OvenMenu`.
+
+!!! note
+    Mojang typically follows a similar structure for all classes except entities. Those are represented by just their names (e.g. `Pig`, `Zombie`, etc.).
 
-* An `Item` called `PowerRing` would be in an `item` package, with a class name of `PowerRingItem`.
-* A `Block` called `NotDirt` would be in a `block` package, with a class name of `NotDirtBlock`.
-* Finally, a `BlockEntity` for a block called `SuperChewer` would be a `block.entity` or `blockentity` package, with a class name of `SuperChewerBlockEntity`.
+Choose One Method from Many
+---------------------------
 
-Appending your class names with what *kind* of object they are makes it easier to figure out what a class is or guess the class for an object.
+There are many methods for performing a certain task: registering an object, listening for events, etc. It's generally recommended to only use a single method to accomplish a given task. While this does improve code formatting, it also avoid any weird interactions or redundancies that may occur (e.g. your event listener executing twice).
```

Comment:
Perhaps this should mention "consistency" as well; stick to one way to do it and be consistent throughout your code.

Reply:
It would help if I remembered that consistency was a word when I originally wrote this.